src/compression.rs

```diff
@@ -1103,31 +1103,26 @@ fn get_zippers(
     descendants_of_node)
 }
 
-/// the complete result of a single step of compression, this is a somewhat expensive data structure
-/// to create.
 #[derive(Debug, Clone)]
-pub struct CompressionStepResult {
-    pub inv: Invention,
+pub struct CompressionStepData {
```

Comment:
This is one of the bigger changes of this commit.

As discussed on Slack, neither duplicating every field in `CompressionStepResult` for train/test nor having two `CompressionStepResult`s seemed like a good solution; the former is verbose and duplicates a lot of code, while the latter ignores the fact that about half of the fields are really shared.

So, instead, I moved the things which aren't shared between train/test into a separate struct, and left the rest.